src/DataValues/DecimalValue.php

```diff
@@ -360,13 +350,20 @@ public function getArrayValue() {
 	}
 
 	/**
-	 * Constructs a new instance of the DataValue from the provided data.
-	 * This can round-trip with @see getArrayValue
+	 * Constructs a new instance from the provided data. Required for @see DataValueDeserializer.
+	 * This is expected to round-trip with @see getArrayValue.
+	 *
+	 * @deprecated since 0.8.3. Static DataValue::newFromArray constructors like this are
+	 *  underspecified (not in the DataValue interface), and misleadingly named (should be named
+	 *  newFromArrayValue). Instead, use DataValue builder callbacks in @see DataValueDeserializer.
```

Comment:
> for callers, that […] is the only information that is relevant.

It should have become clear that I disagree.

> Why do you think it is critical to give the justification in the deprecation message?

Because as a consumer of this I want to know what's going on, and why I have to change code that was just fine for months, possibly years. As a consumer I need to understand the reasoning so I can follow it and properly apply it to my own code. I will not blindly follow an instruction when it does not make sense to me, especially if there is no simple replacement that can be applied with a regular expression, as it is the case here. If there really is a third-party user outside of DataValueDeserializer he should talk to us and we will introduce something with a better name than the current "newFromArray". A `@todo` can not do this. I don't see how a third-party user should become aware of a `@todo` hidden in some otherwise unchanged code.

> […] provided from the perspective of one particular use of this component.

Yes, because there is only one. Secondary users of these methods don't exist, as far as we know.